### ISSUE TITLE
Not able to access the terralist UI

### ISSUE THREAD

**Issue by gowthamshankar05** [terralist/terralist#319]
I have installed terralist in my local system and I'm using self signed certificates for https requests

**the server is stated successfully**
```
{"level":"info","time":"2024-08-28T15:32:39+05:30","message":"Terralist started, listening on port 5758"}
[GIN-debug] Listening and serving HTTPS on :5758
```

**terraform login is successful**
```
Success! Terraform has obtained and saved an API token.

The new API token will be used for any future Terraform command that must make
authenticated requests to localhost:5758.
```

But as soon as the login is successful it is redirecting me to some url where the it is failing
http://localhost:10007/login?state=<somestate>&code=<somecode>==

When I try to access the UI at https://localhost:5758/

I'm getting a blank page with .gitkeep

<img width="1724" alt="Screenshot 2024-08-28 at 3 40 10 PM" src="https://github.com/user-attachments/assets/c10b933f-8454-4d23-8b93-d83da2caa903">

**Comment by valentindeaconu** [comment]
Hello!

> But as soon as the login is successful it is redirecting me to some url where the it is failing
http://localhost:10007/login?state=&code===

This is something different. When Terraform CLI performs the login, it awakes an HTTP server listening on a random port it chooses (in your case, 10007). When the Oauth flow is completed, the Oauth provider redirects the call back to the so-called "redirect URI" the consumer is providing, in this case, Terraform CLI provides "localhost:10007". Since Terraform is listening on this URL, the redirect is successful and Terraform receives the token it needs.

>When I try to access the UI at https://localhost:5758/
>
>I'm getting a blank page with .gitkeep

Regarding this issue, I believe it's because of your self-signed certificates. Are they signed for localhost? How about the `url` (`TERRALIST_URL`) variable? Is it set to `https://localhost:5758`?

**Comment by gowthamshankar05** [comment]
yes the url (TERRALIST_URL) variable? Is it set to https://localhost:5758

**Comment by valentindeaconu** [comment]
How did you compile the binary? Did you use the `task build -- release` command or did you manually built it with `go build`? I suspect that you don't have the webapp built (your `web/dist` directory is empty).

**Comment by valentindeaconu** [comment]
I will close this due to inactivity. Feel free to add a new message if this problem still persists.
